app/mailers/mailer.rb

```diff
@@ -3,11 +3,18 @@ class Mailer < ActionMailer::Base
   default_url_options[:protocol] = Gemcutter::PROTOCOL
 
   def email_reset(user)
-    @user = user
+    @user = User.find(user['id'])
     mail from: Clearance.configuration.mailer_sender,
-         to: user.email,
-         subject: I18n.t(:confirmation,
-           scope: [:clearance, :models, :clearance_mailer],
-           default: "Email address confirmation")
+         to: @user.email,
+         subject: I18n.t('mailer.confirmation_subject',
+           default: 'Please confirm your email address with RubyGems.org')
+  end
+
+  def email_confirmation(user)
+    @user = User.find(user['id'])
```

Comment:
We are [delivering mails with `delayed_job`](https://github.com/rubygems/rubygems.org/pull/1432/files#diff-ca40be4f7d141afd74d2bfb736b37d05R22), it would serialize the `user`. `User` model has overridden[ `encode_with`](https://github.com/rubygems/rubygems.org/blob/master/app/models/user.rb#L98), and after serialization, all we get is
```ruby
{ "id" => id, "handle" => handle }
```
@indirect [has argued](https://github.com/rubygems/rubygems.org/pull/1301#issuecomment-243018553) that
>  background jobs do not serialize the object at all. Instead, pass the object ID, and use ActiveRecord to find a new copy of the object inside the background job before doing operations on it

However, this is not what I am seeing.. [railscast #171](http://railscasts.com/episodes/171-delayed-job?autoplay=true) says
> The way delayed_job works is that it stores the object in database, that means any object you call delay on as well as any argument you use in the method call are going to be serialized into yaml format in the database for it to use in the backgorund..

We are working on [introducing AMS](https://github.com/rubygems/rubygems.org/pull/1464) and I am sure this hack won't be needed then.